doc/source/dev/core-dev/distributing.rst.inc

```diff
@@ -50,39 +50,40 @@ For dependencies it's important to set lower and upper bounds on their
 versions. For *build-time* dependencies, they are specified in
 ``pyproject.toml`` and the versions will *only* apply to the SciPy build
 itself. It's fine to specify either a range or a specific version for a
-dependency like ``wheel`` or ``setuptools``. For NumPy we have to worry
-about ABI compatibility too, hence we specify the version with ``==``
-to the lowest supported version (because NumPy's ABI is backward but not
-forward compatible).
+dependency like ``meson-python`` or ``pybind11``. For NumPy we have to worry
+about ABI compatibility too. However, with NumPy ``>=2.0.0rc1`` backwards
+compatibility is guaranteed as far back as the NumPy ``1.19`` series so
+specification of a lowest supported version of NumPy at build time is no
+longer required in ``pyproject.toml``.
```

Comment:
I think my wording here could be clearer, and I think the increased flexibility in the NumPy `2.x` series means some of what I say may not be strictly true even if it isn't wrong to do.

To be honest, I believe it may be safe to keep the lower bound to `2.0.0rc1` even when `2.1.x` is out, but perhaps also "ok" to use `2.1.0` as a "lower bound" (>=) at that point?

The way I worded this could also imply that you don't need to include NumPy in the list of build deps at all, which is definitely not true.

Reply:
This seems correct and clear enough to me.

> To be honest, I believe it may be safe to keep the lower bound to `2.0.0rc1` even when `2.1.x` is out, but perhaps also "ok" to use `2.1.0` as a "lower bound" (>=) at that point?

That is indeed safe, it doesn't matter. Using `2.1.0` as the lower bound is undesirable; the range being wider is better and more reflective of reality. So if we change it for a next release, it should be `>=2.0.0`. Long term, say in two years, we want build and runtime ranges to be the same, e.g. both `numpy>=2.0.0,<2.6.0`.